### PR TITLE
Add v1 to v3 conversion code and tests for Profile

### PR DESCRIPTION
**Pull request by gunjan5**
## Description

A very special PR. Easily hurt. Please handle with care.

## Todos
- [x] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/upgrade/etcd/conversionv1v3/profile.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	if !ok {
+		return nil, fmt.Errorf("value is not a valid Profile resource")
+	}
+	bk, ok := kvp.Key.(model.ProfileKey)
+	if !ok {
+		return nil, fmt.Errorf("value is not a valid Profile resource key")
+	}
+
+	ap := apiv3.NewProfile()
+	ap.Name = convertName(bk.Name)
```
Comment (caseydavenport) [inline lib/upgrade/etcd/conversionv1v3/profile.go]: We have a function specifically for profile names - might need tweaking since it expects a list.

Comment (caseydavenport) [reply lib/upgrade/etcd/conversionv1v3/profile.go]: https://github.com/projectcalico/libcalico-go/blob/master/lib/upgrade/etcd/conversionv1v3/workloadendpoint.go#L238:6

Comment (caseydavenport) [reply lib/upgrade/etcd/conversionv1v3/profile.go]: It doesn't matter too much, given we're going to ignore anything with `k8s_ns` on the front anyway... but for correctness we should use that function.

File: lib/upgrade/etcd/conversionv1v3/profile_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+			},
+		},
+		v3API: apiv3.Profile{
+			ObjectMeta: v1.ObjectMeta{
+				Name: "nameymcprofilename",
+			},
+			Spec: apiv3.ProfileSpec{
+				Ingress:       []apiv3.Rule{V3InRule1, V3InRule2},
+				Egress:        []apiv3.Rule{V3EgressRule1, V3EgressRule2},
+				LabelsToApply: map[string]string{"thingy": "val", "meep": "", "mop": ""},
```
Comment (caseydavenport) [inline lib/upgrade/etcd/conversionv1v3/profile_test.go]: Could you add the prefix `pncs.` to one of these label keys? Just for more realistic labels.

**Comment by gunjan5** [comment]
Seems like @caseydavenport is traveling today, so @robbrockbank can you take a look at the second commit to see if I've addressed Casey's comments?

**Pull request merged**